core/src/executor/select/mod.rs

```diff
@@ -166,40 +183,52 @@ pub async fn select_with_labels<'a>(
     };
 
     let TableWithJoins { relation, joins } = &table_with_joins;
-    let table = Table::new(relation)?;
-
-    let columns = fetch_columns(storage, table.get_name()).await?;
-    let join_columns = stream::iter(joins.iter())
-        .map(Ok::<_, Error>)
-        .and_then(|join| {
-            let table = Table::new(&join.relation);
-
-            async move {
-                let table = table?;
-                let table_alias = table.get_alias();
-                let table_name = table.get_name();
-
-                let columns = fetch_columns(storage, table_name).await?;
+    let (rows, columns) = match relation {
+        TableFactor::Table { .. } => {
+            let table = Relation::new(relation)?;
+            let columns = fetch_columns(storage, table.get_name()).await?;
+            let columns = Rc::from(columns);
+            (
+                Rows::Table(fetch_blended(storage, table, Rc::clone(&columns)).await?),
+                columns,
+            )
+        }
+        TableFactor::Derived { subquery, alias } => {
+            let (labels, inline_view) = select_with_labels(storage, subquery, None, true).await?;
+            let inline_view = inline_view.try_collect::<Vec<_>>().await?;
+            let labels_rc = Rc::from(labels.to_owned());
+            let labels = Rc::clone(&labels_rc);
+            let rows = inline_view.into_iter().map(move |row| {
+                Ok(BlendContext::new(
+                    &alias.name,
+                    Rc::clone(&labels_rc),
+                    Some(row),
+                    None,
+                ))
+            });
+            (Rows::Derived(stream::iter(rows)), Rc::clone(&labels))
+        }
+    };
```

Comment:
I mainly focused on these code above.
It requires fetching `rows` and `columns` at the same time but it looks... possible to use shared function in `fetch`.

This kind of shape may work. I renamed `fetch_relation` to `fetch_relation_rows`.
```rust
let TableWithJoins { relation, joins } = &table_with_joins;
let columns = fetch_relation_columns(storage, relation).await?;
let columns = Rc::from(columns);
let rows = {
    let columns = Rc::clone(&columns);
        
    fetch_relation_rows(storage, relation, &None)
        .await?
        .map(move |row| {
            let row = Some(row?);
            let columns = Rc::clone(&columns);
            let alias = get_alias(relation)?;
            Ok(BlendContext::new(alias, columns, row, None))
        })
};  
```

Reply:
I tried what you recommended but it was hard to reuse `fetch_relation_columns`.
Below test was failed 😢
```sql
SELECT *                                                                                                                                                                                                                                                                                   FROM (
    SELECT *
    FROM (
        SELECT COUNT(*) AS cnt FROM InnerTable
    ) AS InlineView
) AS InlineView2"
```
If there are more than two nested inline view, it looks like needs some temporary rows to select like,
```rs
let inline_view = inline_view.try_collect::<Vec<_>>().await?;
```

Reply:
I just solved that with recursive labels though there is another bug in index test yet.